lib/active_record/connection_adapters/cockroachdb/schema_statements.rb

```diff
@@ -55,6 +55,37 @@ def primary_key(table_name)
           end
         end
 
+        def primary_keys(table_name)
+          return super unless database_version >= 24_02_02
+
+          query_values(<<~SQL, "SCHEMA")
+            SELECT a.attname
+              FROM (
+                     SELECT indrelid, indkey, generate_subscripts(indkey, 1) idx
+                       FROM pg_index
+                      WHERE indrelid = #{quote(quote_table_name(table_name))}::regclass
+                        AND indisprimary
+                   ) i
+              JOIN pg_attribute a
+                ON a.attrelid = i.indrelid
+               AND a.attnum = i.indkey[i.idx]
+               AND NOT a.attishidden
+             ORDER BY i.idx
+          SQL
+        end
+
+        def column_names_from_column_numbers(table_oid, column_numbers)
+          return super unless database_version >= 24_02_02
+
+          Hash[query(<<~SQL, "SCHEMA")].values_at(*column_numbers).compact
+            SELECT a.attnum, a.attname
+            FROM pg_attribute a
+            WHERE a.attrelid = #{table_oid}
+            AND a.attnum IN (#{column_numbers.join(", ")})
+            AND NOT a.attishidden
+          SQL
```

Comment:
This is not specific to CRDB >= 24.0.2. On 24.1.2 unique constraints on a region by row table will combine the `crdb_region` and column on the dumped schema. Since `attishidden` is not available in CRDB 24.1.2 this seems to do the trick:

```ruby
def column_names_from_column_numbers(table_oid, column_numbers)
  Hash[query(<<~SQL, "SCHEMA")].values_at(*column_numbers).compact
    SELECT a.attnum, a.attname
    FROM pg_attribute a
    WHERE a.attrelid = #{table_oid}
    AND a.attnum IN (#{column_numbers.join(", ")})
    AND a.attisdropped <> 't'
    AND a.attname <> 'crdb_region'
  SQL
end
```

Without `attisdropped <> t`, I see weird `....pg.dropped.18...` strings in the schema dump. All this is happening after upgrade to `Rails 7.1.x` and Cockroach AR Adapter `7.1.x`. I can't reproduce it on Rails `7.0.x`.

When it comes to `unique_constraints`, I also see a problem with indexes with expressions. For example:

```ruby
    t.index "lower(name::STRING) ASC", name: "index_table_name_redacted_on_name_unique", unique: true
```

is becoming:
```ruby
    t.unique_constraint [], name: "index_table_name_redacted_on_name_unique"
```

Reply:
I created an issue with a workaround for `unique_constraint` [here](https://github.com/cockroachdb/activerecord-cockroachdb-adapter/issues/347).

Reply:
@mmalek-sa I'm not targetting `24.0.2` but `24.2.2`, which is the version where the related issue started raising. We do not need this patch before (hence the conditionnal!)